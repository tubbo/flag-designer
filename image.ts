import { generateFlagURL, type FlagParams } from './url.js'

/**
 * Generate a flag SVG from a given set of parameters.
 */
export async function generateFlagImage(params?: Partial<FlagParams>) {
  const url = generateFlagURL(params)
  const response = await fetch(url)

  if (!response.ok) {
    const message = await response.text()

    throw new Error(message)
  }

  return response.blob()
}
