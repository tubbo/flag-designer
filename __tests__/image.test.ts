import { generateFlagImage } from '../image.js'

test('generateFlagImage', async () => {
  const image = await generateFlagImage()

  expect(image.type).toEqual('image/svg+xml;charset=iso-8859-1')
})
