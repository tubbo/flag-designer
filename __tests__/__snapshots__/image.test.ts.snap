// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateFlagImage returns image data when successful 1`] = `
Blob {
  Symbol(kHandle): Blob {},
  Symbol(kLength): 2411,
  Symbol(kType): "image/svg+xml;charset=iso-8859-1",
}
`;
